### PR TITLE
Added arch detection during build

### PR DESCRIPTION
**Pull request by OlivierCazade**

### CODE REVIEW

**Comment by mariomac** [comment]
Mi main concern here is: can we make sure that the machines in the CI pipeline are x86_64?

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ export GOFLAGS=-mod=vendor
 export GO111MODULE=on
 export CGO_ENABLED=0
 export GOOS=linux
-export GOARCH=amd64
+export GOARCH=$(uname -m)
```
Comment (mariomac) [inline Makefile:11]: Maybe I'd just allow overriding it so we can make sure that if e.g. GitHub Actions are run in an ARM machine, they won't break the containers we deploy to Quay.
```Makefile
GOARCH ?= amd64
```

**Comment by jotak** [comment]
@OlivierCazade if I remember correctly, in your PoC you did different changes than this one, right? Is this PR still relevant?

**Comment by OlivierCazade** [comment]
No, I will submit a PR with multi stage for cross compilation, we can close this one.

**Pull request closed without merging**